backend/sftp/sftp.go

```diff
@@ -997,13 +1004,21 @@ func NewFs(ctx context.Context, name, root string, m configmap.Mapper) (fs.Fs, e
 		}
 
 		// If a public key has been specified then use that
-		if pubkeyFile != "" {
-			certfile, err := os.ReadFile(pubkeyFile)
-			if err != nil {
-				return nil, fmt.Errorf("unable to read cert file: %w", err)
+		if pubkeyFile != "" || pubKey != "" {
+			pubKeyRaw := []byte(pubKey)
```

Comment:
Should it use `strconv.Unquote` the same way as is done for `opt.KeyPem` above, which I'm assuming is to handle backslash escaped special characters?

Reply:
I see. This is probably more specific to the PEM-encoded private key, which "should be on a single line with line endings replaced with '\n'" (as documented), that's why that needs the Unquote part. So I guess you can just forget about my suggestion. 😔